### DISCUSSION TITLE
YAML Configuration for Building iOS App on CodeMagic with Fastlane

### DISCUSSION THREAD

**Discussion by bartleby** [codemagic-ci-cd/codemagic-docs#2277] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Fastlane

### Steps to reproduce

I'm trying to build a project for which the source code is not accessible. This project comes with build instructions, and below is a part of it for running the project. The project builds and runs successfully in Xcode, but I am unable to build it on CodeMagic. I cannot figure out how to correctly compose the YAML configuration file.

Here is the relevant section of the instructions to run the project:

Basic Requirements:

iOS app compatibility: iOS 14 and above
Xcode version: 14.3.1 (not higher)
macOS version: 13 and above for optimal Xcode compatibility
External dependencies included in the project archive

Local Setup Steps:

Create or log in to the developer account on developer.apple.com with developer privileges.
Download and install Xcode, ensuring it is moved to the "Applications" folder.
Run the terminal and execute the command: sudo xcode-select -s /Applications/Xcode.app
Unarchive the project archive [integrators-travel-app-ios-main].
Launch the Terminal and run the command: /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" to install the HomeBrew manager.
Execute the command: brew install fastlane
Navigate to the project directory using the command: cd /path/to/travel-app-ios
Run the command: fastlane initialize to install cocoapods, xcodegen, and needle.
Run the command: fastlane setup for subsequent project modifications.
In the event of encountering the prompt "The application 'project_configurator' can't be opened because it is from an unidentified developer," follow these additional steps:

Go to "Settings → Security and Privacy."
Choose "Open Anyway" for the "project_configurator" application.
Repeat the fastlane setup command in the terminal while in the project directory.
Upon completing these steps, the project workspace (travel-app.xcworkspace) is generated, and Xcode can be opened with it.

I have not been able to find a solution in my search.

### Expected results

I expect the project to build successfully.

### Actual results

I have tried specifying these commands in the YAML file, but I encounter an error.

```yaml
  scripts:
    - name: Run fastlane
      script: | 
        bundle install
        bundle exec fastlane initialize
        bundle exec fastlane setup
```

Error:
```
Invalid configuration: 2 validation errors for YamIConfiguration workflows -> ios-workflow -> scripts -> 0 str type expected (type=type_error.str) workflows -> ios-workflow - scripts -> O -> script str type expected (type=type_error.str)
X
Build for avia-new-mvp failed
Invalid configuration: 2 validation errors for YamIConfiguration workflows -> jos-workflow -> scripts
-> O str type expected (type=type_error.str) workflows -> ios-workflow -> scripts -> O -> script str type expected (type=type_error.str)
Click here to open build
```

**Comment by icarusdust** [comment]
Hi @bartleby, it is hard to say where your YAML config is incorrect without seeing the whole `codemagic.yaml` file. So, please share your app id (you can find it in the url: `https://codemagic.io/app/YOUR_APP_ID/` ) or a build id (you can find it after triggering a build under the Build overview section unless the validation error is thrown before triggering a build), so that I can check it for you. Do not share your `codemagic.yaml` here, because I need to see the format that you are using when triggering a build.

**Reply by bartleby** [reply]
hi, APP ID - 657e5c4466561410a78cdc72

**Reply by icarusdust** [reply]
Thanks for the ID. 

The problem is that you are missing `| `after `script`. Your YAML snippet above in the message is correct, but your actual `codemagic.yaml` file does have the required ` |`

**Comment by icarusdust** [comment]
I am closing the thread due to inactivity. Let me know in case of any further questions.
